pkg/apis/garden/types.go

```diff
@@ -910,10 +910,10 @@ type Kube2IAMRole struct {
 	Policy string
 }
 
-// Backup holds information about the backup interval and maximum.
+// Backup holds information about the backup schedule and maximum.
 type Backup struct {
-	// IntervalInSecond defines the interval in seconds how often a backup is taken from etcd.
-	IntervalInSecond int
+	// Schedule defines the cron schedule according to which a backup is taken from etcd.
+	Schedule string
```

Comment:
If they are optional then they should be pointers with `// +optional` tag.

Reply:
The `Backup` struct itself is optional, however, the fields inside are not.

Reply:
Then we should get rid of https://github.com/gardener/gardener/pull/121/files#diff-8221619a3b5265f85b5ec0aad71be59bR34

Reply:
That's for backwards compatibility (we remove `intervalInSecond` and replace it with `schedule`).

Reply:
You can't simply remove existing field and expect that the change to be backwards compatible - some Shoot resources might have been already stored in etcd with `v1beta1` version.

I'll suggest to keep the `IntervalInSecond` in `v1beta1` package for backwards-compatibility, remove it from the internal version, have a conversion for `v1beta1.IntervalInSecond` -> `internal.Schedule` and once `v1beta2` is available, have only `Schedule` there.

Reply:
Agreed that this would be the correct procedure, however, as the `backup` fields weren't used till now because no backup functionality was implemented we decided to go this way.
We can't do the same for fields which already have a meaning, for example `.spec.dns.domain` or what not (in case we would like to change/rename).
The fact that already stored Shoot resources having `intervalInSecond` does not harm, it's just ignored during (un)marshalling.
Well, this was the last part of the Shoot resource API definition that wasn't implemented, therefore let's ease the integration path by doing it as proposed (instead of creating a new API version). In the future we will need to create new versions when doing things like that. OK?

Reply:
Okay. Good enough.